### DISCUSSION TITLE
Best practice for selectively overriding styles

### DISCUSSION THREAD

**Discussion by erikdstock** [saadeghi/daisyui#1705] [Q&A] [answered]
Hello! I'm working on a design system and wondering whether I can reduce the amount of up-front work by adding some styles on top of daisy-ui. Most of the work here seems fine, but occasionally i want to change minor values like padding or font-size. It seems like something like this would mostly work:
```css
.btn { @apply py-3 text-lg; }
```

But that doesn't seem right - it still requires me understanding how the existing `btn` styles work, and I'm potentially importing styles i no longer need. Is this a case where using unstyled daisyUI is the right approach? Or would I be better off just starting from scratch without using this plugin at all?

**Comment by saadeghi** [comment, marked as answer]
>it still requires me understanding how the existing `btn` styles work.

That would be true about any component library (Material, Bootstrap, etc). After all, someone else wrote that styles and if you want to customize it, you should discover what styles are currently used.  
But it's not really hard to handle. For example if you want to customize the padding of all buttons, you can easily look up the browser devtools, you see the button has `1rem` padding, you can add a utility class to change the padding.

>I'm potentially importing styles i no longer need

CSS minifiers can remove unused overridden styles. For example [Postcss-clean](https://github.com/leodido/postcss-clean), [clean-css](https://www.npmjs.com/package/clean-css) or [cssnano](https://cssnano.co/).

>Is this a case where using unstyled daisyUI is the right approach?

Depends on how much design decisions are you going to remove.  
If your target design just needs a few modifications and adding 6~10 utility class to each component will give you what you want, I would say it would be more efficient than starting a new design system from scratch.  
But if what you're trying to create is completely different from daisyUI's design decisions, you might want to turn the `styled` config off which removes the opinionated design decisions from the components. and even if you still found yourself overriding a lot of styles, maybe you need to start creating your own design system from scratch.


Let me know if you have any questions.
